src/vm/i386/cgenx86.cpp

```diff
@@ -299,7 +299,12 @@ void TransitionFrame::UpdateRegDisplay(const PREGDISPLAY pRD)
 
     MethodDesc * pFunc = GetFunction();
     _ASSERTE(pFunc != NULL);
+
+#ifdef WIN64EXCEPTIONS
+    UpdateRegDisplayHelper(pRD, 0);
```

Comment:
I encountered the following assertion failure when I revised the code as you suggested:
```
FAILED   - [ 026 :    5 sec ] GC/API/GC/GetGeneration_fail/GetGeneration_fail.sh
               BEGIN EXECUTION
               /root/dotnet-test/coreclr_unittest_20170208_193938_x86_x86-emu/coreclr-unittest-170208-104248/overlay/corerun GetGeneration_fail.exe
               This test should throw an exception!
               Generation: 0

               Assert failure(PID 1078 [0x00000436], Thread: 1078 [0x0436]): Consistency check failed: FAILED: state.fFound
                   File: /home/parjong/projects/dotnet/coreclr/src/vm/exceptionhandling.cpp Line: 360
                   Image: /root/dotnet-test/coreclr_unittest_20170208_193938_x86_x86-emu/coreclr-unittest-170208-104248/overlay/corerun
```

Here is the state just before that assertion failure happens (inside ``TransitionFrame::UpdateRegDisplayHelper``)
```
TID 1e30: CEHelper::CanMethodHandleException - Processing CorruptionSeverity: 2.
TID 1e30: ....  RESUMEFRAME:  sf is  FFFFA7C8 and  m_sfResumeStackFrame: FFFFA7C8
TID 1e30: ....  RESUMEFRAME:  REACHED initial resume frame: FFFFA7C8
TID 1e30: ....  RESUMEFRAME:  Found last frame to process finallys in, need to process only part of call frame
TID 1e30: ....  RESUMEFRAME:  Limit clause token: EFEB049C
TID 1e30: ....  considering typed clause [b6,e2], ControlPc is inside clause (offset ce)
TID 1e30: ....  found limit clause, stopping clause enumeration
TID 1e30: ....  second pass unwind completed
TID 1e30: ....  calling catch at 0xEFEB5AA2
TID 1e30: ....    calling handler at 0xEFEB5AA2, sp = 0xFFFFA7C8
TID 1e30: ....  resume address should be 0xEFEB5A12
TID 1e30: ....second pass finished
TID 1e30: ....cleaning up ExceptionTracker state
TID 1e30: ....Unlinking ExceptionTracker object 0x08088A14, thread = 0x0808A9F0
TID 1e30: DestroyHandle: *F62F10BC->F2F3F500
TID 1e30: ..TrackerAllocator: freeing tracker 0x08088A14, thread = 0x0808A9F0
TID 1e30: CreateHandle: F62F10BC, type=2
TID 1e30: DestroyHandle: *F62F10C0->F2F3F500
TID 1e30: CEHelper::ResetLastActiveCorruptionSeverityPostCatchHandler - Reset LastActiveException corruption severity to 2.
TID 1e30: ..FinishSecondPass complete, uResumePC = EFEB5A12, current SP = FFFF94BC
TID 1e30: STACKWALK: [000] FILTER  : EXPLICIT : PC= 00000000  SP= 00000000  Frame= ffffa70c  vtbl= f7c5fc98
TID 1e30: STACKWALK: [001] CONSIDER: EXPLICIT : PC= 00000000  SP= 00000000  Frame= ffffa70c  vtbl= f7c5fc98
TID 1e30: STACKWALK: [001] CALLBACK: EXPLICIT : PC= 00000000  SP= 00000000  Frame= ffffa70c  vtbl= f7c5fc98

Breakpoint 1, TransitionFrame::UpdateRegDisplayHelper (this=0xffffa70c, pRD=0xffff8dd8, cbStackPop=12) at /home/parjong/projects/dotnet/coreclr/src/vm/i386/cgenx86.cpp:324
324         CalleeSavedRegisters* regs = GetCalleeSavedRegisters();
(gdb) p/x GetSP()
$1 = 0xffffa758
(gdb) bt 20
#0  TransitionFrame::UpdateRegDisplayHelper (this=0xffffa70c, pRD=0xffff8dd8, cbStackPop=12) at /home/parjong/projects/dotnet/coreclr/src/vm/i386/cgenx86.cpp:324
#1  0xf7126a52 in TransitionFrame::UpdateRegDisplay (this=0xffffa70c, pRD=0xffff8dd8) at /home/parjong/projects/dotnet/coreclr/src/vm/i386/cgenx86.cpp:305
#2  0xf6df7cd6 in StackFrameIterator::NextRaw (this=0xffff89d0) at /home/parjong/projects/dotnet/coreclr/src/vm/stackwalk.cpp:2642
#3  0xf6df32f1 in StackFrameIterator::Next (this=0xffff89d0) at /home/parjong/projects/dotnet/coreclr/src/vm/stackwalk.cpp:1612
#4  0xf6df2a55 in Thread::StackWalkFramesEx (this=0x808a9f0, pRD=0xffff8dd8, pCallback=0xf710e240 <UpdateObjectRefInResumeContextCallback(CrawlFrame*, void*)>, pData=0xffff8d58, flags=0, pStartFrame=0xffffa70c)
    at /home/parjong/projects/dotnet/coreclr/src/vm/stackwalk.cpp:963
#5  0xf710e711 in ExceptionTracker::FindNonvolatileRegisterPointers (pThread=0x808a9f0, uOriginalSP=4294944600, pRegDisplay=0xffff8dd8, uResumeFrameFP=4294944704)
    at /home/parjong/projects/dotnet/coreclr/src/vm/exceptionhandling.cpp:342
#6  0xf7114ee3 in FixNonvolatileRegisters (uOriginalSP=4294944600, pThread=0x808a9f0, pContextRecord=0xffff9f28, fAborting=false) at /home/parjong/projects/dotnet/coreclr/src/vm/exceptionhandling.cpp:1217
#7  0xf7112223 in ProcessCLRException (pExceptionRecord=0x811f52c, MemoryStackFp=4294944712, pContextRecord=0xffff9f28, pDispatcherContext=0xffff9a78)
    at /home/parjong/projects/dotnet/coreclr/src/vm/exceptionhandling.cpp:1136
#8  0xf711c63e in UnwindManagedExceptionPass2 (ex=..., unwindStartContext=0xffff9f28) at /home/parjong/projects/dotnet/coreclr/src/vm/exceptionhandling.cpp:4382
#9  0xf711cfdf in UnwindManagedExceptionPass1 (ex=..., frameContext=0xffffa2b0) at /home/parjong/projects/dotnet/coreclr/src/vm/exceptionhandling.cpp:4536
#10 0xf711da77 in DispatchManagedException (ex=..., isHardwareException=false) at /home/parjong/projects/dotnet/coreclr/src/vm/exceptionhandling.cpp:4643
#11 0xf7298920 in PreStubWorker (pTransitionBlock=0xffffa73c, pMD=0xefec9970) at /home/parjong/projects/dotnet/coreclr/src/vm/prestub.cpp:1078
#12 0xf713440f in ThePreStub () at /home/parjong/projects/dotnet/coreclr/src/vm/i386/asmhelpers.S:921
#13 0xefeb59ff in ?? ()
#14 0xefeb57b3 in ?? ()
#15 0xefeb56b8 in ?? ()
#16 0xefeb553c in ?? ()
#17 0xf267459a in ?? ()
#18 0xf2673b64 in ?? ()
#19 0xf266dfa2 in ?? ()
(More stack frames follow...)
(gdb) info frame 12
Stack frame at 0xffffa758:
 eip = 0xf713440f in ThePreStub (/home/parjong/projects/dotnet/coreclr/src/vm/i386/asmhelpers.S:921); saved eip = 0xefeb59ff
 called by frame at 0xffffa7c8, caller of frame at 0xffffa730
 source language asm.
 Arglist at 0xffffa750, args:
 Locals at 0xffffa750, Previous frame's sp is 0xffffa758
 Saved registers:
  ecx at 0xffffa740, edx at 0xffffa73c, ebx at 0xffffa74c, ebp at 0xffffa750, esi at 0xffffa748, edi at 0xffffa744, eip at 0xffffa754
(gdb) p/x 4294944600
$2 = 0xffffa758
(gdb)
```